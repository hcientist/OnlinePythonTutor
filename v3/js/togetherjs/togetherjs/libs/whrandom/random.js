(function () {

function RandomStream(seed) {
  if (! seed) {
    seed = Date.now();
  }
  if (typeof seed != "number") {
    seed = parseInt(seed, 10);
  }
  if ((! seed) || isNaN(seed)) {
    throw "Bad seed";
  }
  var x = (seed % 30268) + 1;
  seed = (seed - (seed % 30268)) / 30268;
  var y = (seed % 30306) + 1;
  seed = (seed - (seed % 30306)) / 30306;
  var z = (seed % 30322) + 1;
  seed = (seed - (seed % 30322)) / 30322;
  return function random() {
    x = (171 * x) % 30269;
    y = (172 * y) % 30307;
    z = (170 * z) % 30323;
    if (random.logState) {
      console.log('x', x, 'y', y, 'z', z);
    }
    return (x / 30269.0 + y / 30307.0 + z / 30323.0) % 1.0;
  };
}

// Support for AMD/requirejs, CommonJS/Node, and no module system:
if (typeof define != "undefined") {
  define([], function () {return RandomStream;});
} else if (typeof module != "undefined") {
  module.exports = RandomStream;
} else {
  window.WHRandomStream = RandomStream;
}

})();
