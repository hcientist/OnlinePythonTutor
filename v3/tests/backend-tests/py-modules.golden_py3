{
  "code": "# test to make sure nothing crashes when tracing code in pure-Python\n# function calls imported from modules\nimport random\nrandom.randint(10, 100)\n\nfrom re import compile\nx = compile('foo')\n", 
  "trace": [
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 3, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "random"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "random": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "INSTANCE", 
          "module"
        ]
      }, 
      "line": 4, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "random"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "random": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "INSTANCE", 
          "module"
        ]
      }, 
      "line": 6, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "random", 
        "compile"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "compile": [
          "IMPORTED_FAUX_PRIMITIVE", 
          "imported object"
        ], 
        "random": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "INSTANCE", 
          "module"
        ]
      }, 
      "line": 7, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "random", 
        "compile", 
        "x"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "compile": [
          "IMPORTED_FAUX_PRIMITIVE", 
          "imported object"
        ], 
        "x": [
          "REF", 
          2
        ], 
        "random": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "INSTANCE", 
          "module"
        ], 
        "2": [
          "INSTANCE", 
          "SRE_Pattern"
        ]
      }, 
      "line": 7, 
      "event": "return"
    }
  ]
}
