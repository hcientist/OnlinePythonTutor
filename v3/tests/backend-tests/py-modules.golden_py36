{
  "code": "# test to make sure nothing crashes when tracing code in pure-Python\n# function calls imported from modules\nimport random\nrandom.randint(10, 100)\n\nfrom re import compile\nx = compile('foo')\n",
  "trace": [
    {
      "line": 3,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": ""
    },
    {
      "line": 4,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "random": [
          "REF",
          1
        ]
      },
      "ordered_globals": [
        "random"
      ],
      "stack_to_render": [],
      "heap": {
        "1": [
          "INSTANCE",
          "module"
        ]
      },
      "stdout": ""
    },
    {
      "line": 6,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "random": [
          "REF",
          1
        ]
      },
      "ordered_globals": [
        "random"
      ],
      "stack_to_render": [],
      "heap": {
        "1": [
          "INSTANCE",
          "module"
        ]
      },
      "stdout": ""
    },
    {
      "line": 7,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "random": [
          "REF",
          1
        ],
        "compile": [
          "IMPORTED_FAUX_PRIMITIVE",
          "imported object"
        ]
      },
      "ordered_globals": [
        "random",
        "compile"
      ],
      "stack_to_render": [],
      "heap": {
        "1": [
          "INSTANCE",
          "module"
        ]
      },
      "stdout": ""
    },
    {
      "line": 7,
      "event": "return",
      "func_name": "<module>",
      "globals": {
        "random": [
          "REF",
          1
        ],
        "compile": [
          "IMPORTED_FAUX_PRIMITIVE",
          "imported object"
        ],
        "x": [
          "REF",
          2
        ]
      },
      "ordered_globals": [
        "random",
        "compile",
        "x"
      ],
      "stack_to_render": [],
      "heap": {
        "1": [
          "INSTANCE",
          "module"
        ],
        "2": [
          "INSTANCE_PPRINT",
          "SRE_Pattern",
          "re.compile('foo')"
        ]
      },
      "stdout": ""
    }
  ]
}
