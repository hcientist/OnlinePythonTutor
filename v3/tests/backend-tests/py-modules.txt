# test to make sure nothing crashes when tracing code in pure-Python
# function calls imported from modules
import random
random.randint(10, 100)

from re import compile
x = compile('foo')
