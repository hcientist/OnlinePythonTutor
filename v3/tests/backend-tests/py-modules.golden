{
  "code": "# test to make sure nothing crashes when tracing code in pure-Python\n# function calls imported from modules\nimport random\nrandom.randint(10, 100)\n\nfrom re import compile\nx = compile('foo')\n", 
  "trace": [
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 3, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "random"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "random": [
          "IMPORTED_FAUX_PRIMITIVE", 
          "imported module"
        ]
      }, 
      "heap": {}, 
      "line": 4, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "random"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "random": [
          "IMPORTED_FAUX_PRIMITIVE", 
          "imported module"
        ]
      }, 
      "heap": {}, 
      "line": 6, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "random", 
        "compile"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "compile": [
          "IMPORTED_FAUX_PRIMITIVE", 
          "imported function"
        ], 
        "random": [
          "IMPORTED_FAUX_PRIMITIVE", 
          "imported module"
        ]
      }, 
      "heap": {}, 
      "line": 7, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "random", 
        "compile", 
        "x"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "compile": [
          "IMPORTED_FAUX_PRIMITIVE", 
          "imported function"
        ], 
        "x": [
          "REF", 
          1
        ], 
        "random": [
          "IMPORTED_FAUX_PRIMITIVE", 
          "imported module"
        ]
      }, 
      "heap": {
        "1": [
          "_sre.SRE_Pattern", 
          "<_sre.SRE_Pattern object at 0xADDR>"
        ]
      }, 
      "line": 7, 
      "event": "return"
    }
  ]
}
